### PR TITLE
Rename file to file share

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

**Comment by joshfree** [comment]
Fixes #5909

File: sdk/storage/azure-storage-file-share/README.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ To make this possible you'll need the Account SAS (shared access signature) stri
 ## Key concepts
 
 ### URL format
-Files are addressable using the following URL format:
+File Share are addressable using the following URL format:
```
Comment (joshfree) [inline sdk/storage/azure-storage-file-share/README.md]: Missing S: File Shares are addressable

File: sdk/storage/azure-storage-file-share/README.md [hunk 2]
```diff
@@ -167,21 +167,21 @@ The URI to reference a share, directory or file must be unique. Within a given s
 If you attempt to create a share, directory, or file with a name that violates naming rules, the request will fail with status code 400 (Bad Request).
 
 ### Share Names
-The rules for File service share names are more restrictive than what is prescribed by the SMB protocol for SMB share names, so that the Blob and File services can share similar naming conventions for containers and shares. The naming restrictions for shares are as follows:
+The rules for File Share service share names are more restrictive than what is prescribed by the SMB protocol for SMB share names, so that the Blob and File services can share similar naming conventions for containers and shares. The naming restrictions for shares are as follows:
```
Comment (joshfree) [inline sdk/storage/azure-storage-file-share/README.md]: Should be “The rules for File Share service names”

File: sdk/storage/azure-storage-file-share/README.md [hunk 3]
```diff
@@ -167,21 +167,21 @@ The URI to reference a share, directory or file must be unique. Within a given s
...
 
 1. A share name must be a valid DNS name.
 1. Share names must start with a letter or number, and can contain only letters, numbers, and the dash (-) character.
 1. Every dash (-) character must be immediately preceded and followed by a letter or number; consecutive dashes are not permitted in share names.
 1. All letters in a share name must be lowercase.
 1. Share names must be from 3 through 63 characters long.
 
-### Directory and File Names
-The Azure File service naming rules for directory and file names are as follows:
+### Share Directory and File Names
```
Comment (joshfree) [inline sdk/storage/azure-storage-file-share/README.md]: Either change Share to File Share or just revert this change

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileSasPermission.java [hunk 4]
```diff
@@ -11,10 +11,10 @@
  * This is a helper class to construct a string representing the permissions granted by a ServiceSAS to a file. Setting
  * a value to true means that any SAS which uses these permissions will grant permissions for that operation. Once all
  * the values are set, this should be serialized with toString and set as the permissions field on a {@link
- * FileServiceSasSignatureValues} object. It is possible to construct the permissions string without this class, but the
- * order of the permissions is particular and this class guarantees correctness.
+ * ShareServiceSasSignatureValues} object. It is possible to construct the permissions string without this class, but
+ * the order of the permissions is particular and this class guarantees correctness.
  */
```
Comment (joshfree) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileSasPermission.java]: @alzimmermsft PTAL wrt SAS changes

**Comment by alzimmermsft** [comment]
@sima-zhu @joshfree This has priority over the other SAS changes as this has modifications to Azure Core.

@amishra-dev Could we prioritize a review from the Storage team side given the modifications to Azure Core, thank you!

File: sdk/storage/azure-storage-file-share/README.md (hunk 1 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file-share/README.md]: Done

File: sdk/storage/azure-storage-file-share/README.md (hunk 2 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file-share/README.md]: Done.

File: sdk/storage/azure-storage-file-share/README.md (hunk 3 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file-share/README.md]: Revert it back as some of other places refer to directory and file only.
